jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -243,10 +304,24 @@ def collapse_to_spectrum(self, add_data=True, **kwargs):
         # is always wavelength. This may need adjustment after the following
         # specutils PR is merged: https://github.com/astropy/specutils/pull/1033
         spatial_axes = (0, 1)
-
-        collapsed_nddata = getattr(nddata_reshaped, self.function_selected.lower())(
-            axis=spatial_axes, **kwargs
-        )  # returns an NDDataArray
+        if self.function_selected.lower() == 'mean':
+            # Use built-in sum function to collapse NDDataArray
+            collapsed_for_mean = nddata_reshaped.sum(axis=spatial_axes, **kwargs)
+            # But we still need the mean function for everything except flux
+            collapsed_as_mean = nddata_reshaped.mean(axis=spatial_axes, **kwargs)
+
+            # Then normalize the flux based on the fractional pixel array
```

Comment:
I was hoping all this would happen inside NDData somewhere. Kinda defeats the purpose of using NDData if we still have to do it ourselves here, no, @bmorris3 ?